library/core/class.applicationmanager.php

```diff
@@ -48,35 +48,26 @@ public function __construct(AddonManager $addonManager = null) {
      */
     public function availableApplications() {
         if (!is_array($this->availableApplications)) {
-            $ApplicationInfo = array();
-
-            $AppFolders = Gdn_FileSystem::folders(PATH_APPLICATIONS); // Get an array of all application folders
-            // Now look for about files within them.
-            $ApplicationAboutFiles = Gdn_FileSystem::findAll(PATH_APPLICATIONS, 'settings'.DS.'about.php', $AppFolders);
-            // Include them all right here and fill the application info array
-            $ApplicationCount = count($ApplicationAboutFiles);
-            for ($i = 0; $i < $ApplicationCount; ++$i) {
-                include($ApplicationAboutFiles[$i]);
-
-                // Define the folder name for the newly added item
-                foreach ($ApplicationInfo as $ApplicationName => $Info) {
-                    if (array_key_exists('Folder', $ApplicationInfo[$ApplicationName]) === false) {
-                        $Folder = substr($ApplicationAboutFiles[$i], strlen(PATH_APPLICATIONS));
-                        if (substr($Folder, 0, 1) == DS) {
-                            $Folder = substr($Folder, 1);
-                        }
-
-                        $Folder = substr($Folder, 0, strpos($Folder, DS));
-                        $ApplicationInfo[$ApplicationName]['Folder'] = $Folder;
-                    }
+            $applications = [];
+
+            $addons = $this->addonManager->lookupAllByType(Addon::TYPE_ADDON);
+            foreach ($addons as $addon) {
+                /* @var Addon $addon */
+                if ($addon->getInfoValue('oldType') !== 'application') {
+                    continue;
```

Comment:
Right away you have some issues here. You need to make Gdn_ApplicationManager backwards compatible or else some plugin somewhere is going to break. Look at how Gdn_PluginManager does it:

```php
public function availablePlugins() {
    $addons = $this->addonManager->lookupAllByType(Addon::TYPE_ADDON);

    $result = [];
    foreach ($addons as $addon) {
        /* @var Addon $addon */
        if ($addon->getInfoValue('oldType') !== 'plugin') {
            continue;
        }

        $infoArray  = static::calcOldInfoArray($addon);
        $result[$infoArray['Index']] = $infoArray;
    }
    return $result;
}
```

You can do something almost exactly the same and use Gdn_PluginManager::calcOldInfoArray() too.